### DISCUSSION TITLE
[Idea]  Change practice on setting max ps_versions_compliancy

### DISCUSSION THREAD

**Discussion by ChillCode** [PrestaShop/PrestaShop#34589] [Ideas]
### The problem

Hi,

Almost all modules set version compliancy like this:

`$this->ps_versions_compliancy = array('min' => '1.7.8.0', 'max' => _PS_VERSION_);`

So I understand that the module will be always compatible with the last release, but some times is not true, and while upgrading a site to check compatibility with 8.1.2 I faced upgrade issues because of third-party modules, they are not really compatible.

Is this correct or I'm missing something?

### The solution

Lock the max to the current major version?

$this->ps_versions_compliancy = array('min' => '1.7.8.0', 'max' => '1.7.8');

### Additional context

An issue related to this topic, I know the module is not maintained anymore:

https://github.com/PrestaShop/PrestaShop/issues/34579

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
hi @ChillCode 

we've had a heated discussion on this subject recently with other members of the technical council of the project (cc @Hlavtox ;))

I agree with you that we should be more strict and require developers not to use _PS_VERSION_ here. One legitimate issue that @Hlavtox raised is that even if you don't need any changes in the module, you'd have to change the version compliance in every single module you maintain. For us, it would mean that we need to edit over 60 modules and change, for example, `8`, to `9` and it doesn't sound good.

We wondered with @MatShir if it wouldn't be a good idea to force this rule only on [the official marketplace](https://addons.prestashop.com/en/), but that's not all, that would also require us to edit the upgrade process and see whether the modules are compatible with the target version or not.

**Reply by ChillCode** [reply]
Thanks for the clarification @kpodemski !
